Tests/SentryTests/Profiling/SentryProfilerSwiftTests.swift

```diff
@@ -0,0 +1,373 @@
+import Sentry
+import XCTest
+
+#if os(iOS) || os(macOS) || targetEnvironment(macCatalyst)
+class SentryProfilerSwiftTests: XCTestCase {
+    private static let dsnAsString = TestConstants.dsnAsString(username: "SentryProfilerSwiftTests")
+
+    private class Fixture {
+        lazy var options: Options = {
+            let options = Options()
+            options.dsn = SentryProfilerSwiftTests.dsnAsString
+            return options
+        }()
+        lazy var client: TestClient! = TestClient(options: options)
+        lazy var hub: SentryHub = {
+            let hub = SentryHub(client: client, andScope: scope)
+            hub.bindClient(client)
+            Dynamic(hub).tracesSampler.random = TestRandom(value: 1.0)
+            Dynamic(hub).profilesSampler.random = TestRandom(value: 0.5)
+            return hub
+        }()
+        let scope = Scope()
+        let message = "some message"
+        let transactionName = "Some Transaction"
+        let transactionOperation = "Some Operation"
+    }
+
+    private var fixture: Fixture!
+
+    override func setUp() {
+        super.setUp()
+        fixture = Fixture()
+        SentryTracer.resetAppStartMeasurementRead()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        clearTestState()
+        SentryTracer.resetAppStartMeasurementRead()
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+        SentryFramesTracker.sharedInstance().resetFrames()
+        SentryFramesTracker.sharedInstance().stop()
+#endif
+    }
+
+    // This test is only available on newer APIs because of the availability of `DispatchQueue.activate`
+    @available(tvOS 10.0, *)
+    @available(OSX 10.12, *)
+    @available(iOS 10.0, *)
+    func testConcurrentProfilingTransactions() {
+        let options = fixture.options
+        options.profilesSampleRate = 1.0
+        options.tracesSampleRate = 1.0
+
+        let queue = DispatchQueue(label: "SentryProfilerSwiftTests", attributes: [.concurrent, .initiallyInactive])
+        let group = DispatchGroup()
+
+        let numberOfTransactions = 10
+        for _ in 0 ..< numberOfTransactions {
+            group.enter()
+            let exp = expectation(description: "finished span")
+            let span = fixture.hub.startTransaction(name: fixture.transactionName, operation: fixture.transactionOperation)
```

Comment:
`l`
```suggestion
            let transaction = fixture.hub.startTransaction(name: fixture.transactionName, operation: fixture.transactionOperation)
```

Reply:
In my reading, `SentryTransaction` is the thing we use to package up information about a `SentrySpanProtocol` conforming instance for transmission to our backend, not something we use in ongoing tracking. Since `startTransaction` returns a `SentrySpan`, that's why I used `span` here. (Honestly we should change the name of the methods using `transaction` to use `span` instead, but I understand that's a breaking change we can't do right now).

Reply:
No strong opinions on this one.